### ISSUE TITLE
BatchNormalization Layers work diffrently from Python TensorFlow

### ISSUE THREAD

**Issue by edge7** [tensorflow/tfjs#7250]
**System information**
- TensorFlow.js version: 4.2.0
- Are you willing to contribute it: Yes

**Describe the feature and the current behavior/state.**
In Python it is possible to create some layers like this:
`tf.keras.layers.BatchNormalization()(f, training=True)
`
an other example is:
`u =tf.keras.layers.Dropout(0.3)(u, training=True)
`
Like that, the call method of the Layer will get the kwargs training=True always, also at prediction time without needing of
calling the model like this:
`model(input, training=True)
`
When converting from Python TF to TFJS, the models does not seem to work like that, so the user is forced to call the model like this:
`await this.loaded_model.apply(batched, {'training': true});
`
as calling it like this:
`this.loaded_model.predict(batched)
`
does not work as expected (i.e.: the batchNormLayer still works in inference mode).
Also, models, where at least one layer has the above signature, are not convertible in graph_mode

**Who will benefit with this feature?**
People that use GAN with TFJS, as for most of the GAN models, batch normalization needs to work in training mode when predicting.

**Any Other info.**
Please comment if my point is not clear.

**Comment by pyu10055** [comment]
hi @edge7 This is a known error for TensorFlow as tracked [here](https://github.com/tensorflow/tensorflow/issues/31668)
The fix seems to be set the trainable to false on the batch normalization layers before you export to saved model. 
Can you give it a try? thanks

**Comment by Varun-S10** [comment]
Hi @edge7, Thank you for your patience. We truly appreciate you flagging this issue, we will file a request internally.

**Comment by Varun-S10** [comment]
Hi @edge7, Thank you for bringing this to our attention and for providing a comprehensive overview. We appreciate the time you invested in creating this issue.

Currently, our internal development team is concentrated on other scheduled milestones, so we cannot prioritize this task immediately. However, we would love to see this move forward as a community-led effort.

If you’d like to submit a fix, please fork the repo and open a PR following the [TensorFlow Contribution Guidelines](https://www.tensorflow.org/community/contribute), [Tfjs Contributions ](https://github.com/tensorflow/tfjs/blob/master/tfjs-vis/CONTRIBUTING.md#how-to-contribute)and be sure to link this issue in your PR to avoid duplicate work, kindly leave a comment here if you plan to take this on.

Thank you again for your interest and time.

**Comment by Varun-S10** [comment]
Hi @edge7, Thank you for opening this issue. Since this issue has been open for a long time, we are closing this issue for now.
If this is still relevant and you would like to work on it, please feel free to reopen the issue or submit a pull request referencing it. Thank you for your understanding.
